src/services/infra/InfraService.ts

```diff
@@ -226,23 +228,28 @@ export default class InfraService {
       )
 
       // Get DNS records from Amplify
-      /**
-       * note: we wait for ard 90 sec as there is a time taken
-       * for amplify to generate the certification manager in the first place
-       * This is a dirty workaround for now, and will cause issues when we integrate
-       * this directly within the Isomer CMS.
-       * todo: push this check into a queue-like system when integrating this with cms
-       */
-      await new Promise((resolve) => setTimeout(resolve, 90000))
-
-      /**
-       * todo: add some level of retry logic if get domain association command
-       * does not contain the DNS redirections info.
-       */
+      const isTestEnv =
+        config.get("env") === "test" || config.get("env") === "dev"
+      // since we mock values during development, we don't have to await for the dns records
+      if (!isTestEnv) {
+        /**
+         * note: we wait for ard 90 sec as there is a time taken
+         * for amplify to generate the certification manager in the first place
+         * This is a dirty workaround for now, and will cause issues when we integrate
+         * this directly within the Isomer CMS.
+         * todo: push this check into a queue-like system when integrating this with cms
+         */
+        await new Promise((resolve) => setTimeout(resolve, 90000))
```

Comment:
This is going to block the main thread for 90s right?

Seems like "todo: push this check into a queue-like system when integrating this with cms" is crucial for integration with CMS

Reply:
hmm correct me if I am wrong, but set setTimeout does [not block the main thread right](https://developer.mozilla.org/en-US/docs/Web/API/setTimeout)?

Reply:
@kishore03109 hmm, you can do a simple test on console -
```
async function test() { if(true) { await new Promise((resolve) => setTimeout(resolve, 2000)) } console.log("DONE") }
```

"DONE" wont be printed until the promise is done. you might want to double check the actual behaviour on this

Reply:
Hmm wait isnt that the point? I want to only resolve this after 90 seconds to give Amplify enough time to generate the DNS results

eg for the code
"""
async function test() {
  await new Promise((resolve) => setTimeout(resolve, 10000))
  console.log("PROMISE DONE")
}
console.log("STARTING")
test()
console.log("ENDED")
"""

in the console log, we would see:
STARTING
ENDED
PROMISE DONE

thus the other functions in the thread will not be blocked?

Reply:
think the confusion stems from the `await` - if you `await` in an async function: 

```ts
const x = await work()
doOtherWork()
const y = x
```

the main thread is free to continue on until the result of the `await`-ed promise is used (ie, `const y = x`). however, because we `await` here, i think we'll pause execution for 90s

Reply:
@kishore03109 if you added `await` on test:

```
console.log("STARTING")
await test()
console.log("ENDED")
```

it will wait for promise to be done first

Reply:
@harishv7, for ease of mind, I have made the main caller (`this.handleSiteLaunchResults`) function non-blocking (ie, removing the `await`), since we technically can run this async after returning the 200 response. However, this means that any error thrown by this will not be caught by `handleSiteLaunchFormRequest`. Therefore, I am wrapping a try catch around and logging the error to any unintended effects of some unexpected error. Ideally, during an error, this would return something to the user, but I think this can be revisited when we are integrating the features with the frontend!

Reply:
i think this is non-blocking - top-most caller is `handleSiteLaunchFormRequest` in `formsgSiteLaunch` and it calls `this.handleSiteLaunchResults(formResponses, submissionId)` without `await` (which calls this method later on)